dateparser/utils.py

```diff
@@ -86,3 +87,37 @@ def find_date_separator(format):
     m = re.search(r'(?:(?:%[dbBmaA])(\W))+', format)
     if m:
         return m.group(1)
+
+
+class Registry(object):
+
+    _global_dict = {}
+    _tag = 'from_registry'
+
+    def __new__(cls, *args, **kwargs):
+        key = cls.get_key(*args, **kwargs)
+
+        if key in cls._global_dict:
+            return cls._tag_instance(cls._global_dict[key])
+
+        return cls._global_dict.setdefault(
+            key,
+            object.__new__(cls, *args, **kwargs)
+        )
```

Comment:
We want to be able to identify when an instance is returned from dictionary and when it is created fresh.  In case,  instance comes from dictionary we prevent `__init__` from being called on it. We manage this by tagging instances.  If we'd tag the instance after it is created and not at the time of returning from dictionary,  it would always appear to be coming  from dictionary and hence `__init__` won't be applied to it.